plugins/database/postgresql/postgresql_test.go

```diff
@@ -151,44 +225,29 @@ func TestPostgreSQL_CreateUser(t *testing.T) {
 
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
-			usernameConfig := dbplugin.UsernameConfig{
-				DisplayName: "test",
-				RoleName:    "test",
-			}
-
-			statements := dbplugin.Statements{
-				Creation: test.createStmts,
-			}
-
 			// Give a timeout just in case the test decides to be problematic
 			ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 			defer cancel()
 
-			username, password, err := db.CreateUser(ctx, statements, usernameConfig, time.Now().Add(time.Minute))
-			if err != nil {
-				t.Fatalf("err: %s", err)
+			resp, err := db.NewUser(ctx, test.req)
+			if test.expectErr && err == nil {
+				t.Fatalf("err expected, got nil")
 			}
-
-			if !test.shouldTestCredsExist {
-				// We're done here.
-				return
+			if !test.expectErr && err != nil {
+				t.Fatalf("no error expected, got: %s", err)
 			}
 
-			if err = testCredsExist(t, db.ConnectionURL, username, password); err != nil {
-				t.Fatalf("Could not connect with new credentials: %s", err)
-			}
+			test.credsAssertion(t, db.ConnectionURL, resp.Username, test.req.Password)
 
 			// Ensure that the role doesn't expire immediately
 			time.Sleep(2 * time.Second)
 
-			if err = testCredsExist(t, db.ConnectionURL, username, password); err != nil {
-				t.Fatalf("Could not connect with new credentials: %s", err)
-			}
+			test.credsAssertion(t, db.ConnectionURL, resp.Username, test.req.Password)
 		})
 	}
 }
 
-func TestPostgreSQL_RenewUser(t *testing.T) {
+func TestUpdateUser_expiration(t *testing.T) {
```

Comment:
Why are there two expiration tests? This one and `TestUpdateUser_Expiration()` on line 409.

Reply:
I'm also curious on this one